nncf/common/pruning/operations.py

```diff
@@ -223,6 +225,63 @@ def mask_propagation(cls, node: NNCFNode, graph: NNCFGraph,
         node.data['output_mask'] = result_mask
 
 
+class SplitPruningOp(BasePruningOp):
+    @classmethod
+    def match_multiple_output_masks(cls, output_masks: List[SymbolicMask], output_edges: List[NNCFGraphEdge],
+                                    chunk_axis: int) -> Dict['str', SymbolicMask]:
+        """
+        Match multiple input mask to each next nodes.
+
+        :param output_masks: Given output masks.
+        :param output_edges: Given output edges of the node.
+        :param chunk_axis: Given the axis on which operation was performed.
+        :return: Matched output mask for each next node.
+        """
+        result_masks = {}
+        tmp_output_masks = output_masks.copy()
+        tmp_output_masks_shape = [mask.shape[0] for mask in tmp_output_masks]
+        for edge in output_edges:
+            node_name = edge.to_node.node_name
+            idx = tmp_output_masks_shape.index(edge.tensor_shape[chunk_axis])
+            result_masks[node_name] = tmp_output_masks.pop(idx)
+            tmp_output_masks_shape.pop(idx)
+
+        return result_masks
+
+    @classmethod
+    def accept_pruned_input(cls, node: NNCFNode):
+        return True
+
+    @classmethod
+    def generate_output_masks(cls, node: NNCFNode, graph: NNCFGraph,
+                              tensor_processor: Type[NNCFPruningBaseTensorProcessor]) -> Optional[NNCFTensor]:
+        """
+        Generate output mask from input masks for split/chunk operations.
+        If input mask is None return None
+
+        :param node: Node to determine its sources.
+        :param graph: NNCF graph to work with.
+        :param tensor_processor: Interface with tensor processing methods.
+        :return: Filled input masks.
+        """
+        input_mask = get_input_masks(node, graph)[0]
```

Comment:
Last comment from my side:
let's check input masks exist before taking it
```suggestion
        input_masks = get_input_masks(node, graph)
        if not input_masks:
            return None
 
        input_mask = input_masks[0]
```